src/main/java/core/basesyntax/strategy/impl/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.strategy.mapvalidator.KeyPresentValidator;
+import java.util.Map;
+
+public class PurchaseOperationHandler
+        extends AbstractOperationHandler
+        implements KeyPresentValidator {
+    public PurchaseOperationHandler() {
+        super(Operation.PURCHASE);
+    }
+
+    @Override
+    protected void processTransaction(Map<String, Integer> fruitQuantityMap,
+                                      FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        validateMap(fruitQuantityMap, transaction);
+        int newQuantity = fruitQuantityMap.get(fruit) - transaction.getQuantity();
+        validateNewQuantity(newQuantity, transaction);
+        fruitQuantityMap.put(fruit, newQuantity);
+    }
+
+    private static void validateNewQuantity(int newQuantity, FruitTransaction transaction) {
+        if (newQuantity < 0) {
+            String exceptionMessage = "Transaction = ["
+                    + transaction + "] resulted in negative quantity of product";
+            throw new IllegalArgumentException(exceptionMessage);
```

Comment:
According to the checklist, we should throw a RuntimeException instead of IllegalArgumentException when the balance is negative after a purchase operation. Make sure to include the relevant transaction information in the exception message for clarity.